src/topology/metric_space/hausdorff_distance.lean

```diff
@@ -866,29 +874,136 @@ lemma thickening_subset_cthickening_of_le {δ₁ δ₂ : ℝ} (hle : δ₁ ≤ 
 end
 
-lemma closure_eq_Inter_cthickening {E : set α} :
+lemma cthickening_eq_Inter_cthickening {δ : ℝ} (δ_nn : 0 ≤ δ) (E : set α) :
+  cthickening δ E = ⋂ (ε : ℝ) (h : δ < ε), cthickening ε E :=
+begin
+  apply cthickening_eq_Inter_cthickening' δ_nn (Ioi δ) rfl.subset,
+  simp_rw inter_eq_right_iff_subset.mpr Ioc_subset_Ioi_self,
+  exact λ _ hε, nonempty_Ioc.mpr hε,
+end
+
+lemma cthickening_eq_Inter_thickening' {δ : ℝ} (δ_nn : 0 ≤ δ)
+  (s : set ℝ) (hsδ : s ⊆ Ioi δ) (hs : ∀ ε, δ < ε → (s ∩ (Ioc δ ε)).nonempty) (E : set α) :
+  cthickening δ E = ⋂ ε ∈ s, thickening ε E :=
+begin
+  apply le_antisymm,
+  { apply subset_bInter,
+    intros ε hε,
+    rcases hs ε (mem_Ioi.mp (hsδ hε)) with ⟨ε', ⟨hsε', hε'⟩⟩,
+    have ss := cthickening_subset_thickening' (lt_of_le_of_lt δ_nn hε'.1) hε'.1 E,
+    exact ss.trans (thickening_mono hε'.2 E), },
+  { rw cthickening_eq_Inter_cthickening' δ_nn s hsδ hs E,
+    apply bInter_mono,
+    exact λ ε hε, thickening_subset_cthickening ε E, },
+end
+
+lemma cthickening_eq_Inter_thickening {δ : ℝ} (δ_nn : 0 ≤ δ) (E : set α) :
+  cthickening δ E = ⋂ (ε : ℝ) (h : δ < ε), thickening ε E :=
+begin
+  apply cthickening_eq_Inter_thickening' δ_nn (Ioi δ) rfl.subset,
+  simp_rw inter_eq_right_iff_subset.mpr Ioc_subset_Ioi_self,
+  exact λ _ hε, nonempty_Ioc.mpr hε,
+end
+
+/-- The closure of a set equals the intersection of its closed thickenings of positive radii
+accumulating at zero. -/
+lemma closure_eq_Inter_cthickening' (E : set α)
+  (s : set ℝ) (hs₀ : s ⊆ Ioi 0) (hs : ∀ ε, 0 < ε → (s ∩ (Ioc 0 ε)).nonempty) :
+  closure E = ⋂ δ ∈ s, cthickening δ E :=
+by { rw ← cthickening_zero, apply cthickening_eq_Inter_cthickening' rfl.ge _ hs₀ hs, }
+
+/-- The closure of a set equals the intersection of its closed thickenings of positive radii. -/
+lemma closure_eq_Inter_cthickening (E : set α) :
   closure E = ⋂ (δ : ℝ) (h : 0 < δ), cthickening δ E :=
-by { rw ← cthickening_zero, exact cthickening_eq_Inter_cthickening rfl.ge, }
+by { rw ← cthickening_zero, exact cthickening_eq_Inter_cthickening rfl.ge E, }
+
+/-- The closure of a set equals the intersection of its open thickenings of positive radii
+accumulating at zero. -/
+lemma closure_eq_Inter_thickening' (E : set α)
+  (s : set ℝ) (hs₀ : s ⊆ Ioi 0) (hs : ∀ ε, 0 < ε → (s ∩ (Ioc 0 ε)).nonempty) :
+  closure E = ⋂ δ ∈ s, thickening δ E :=
+by { rw ← cthickening_zero, apply cthickening_eq_Inter_thickening' rfl.ge _ hs₀ hs, }
+
+/-- The closure of a set equals the intersection of its (open) thickenings of positive radii. -/
+lemma closure_eq_Inter_thickening (E : set α) :
+  closure E = ⋂ (δ : ℝ) (h : 0 < δ), thickening δ E :=
+by { rw ← cthickening_zero, exact cthickening_eq_Inter_thickening rfl.ge E, }
+
+/-- The frontier of the (open) thickening of a set is contained in an `inf_edist` level set. -/
+lemma frontier_thickening_subset (E : set α) {δ : ℝ} (δ_pos : 0 < δ) :
+  frontier (thickening δ E) ⊆ {x : α | inf_edist x E = ennreal.of_real δ } :=
```

Comment:
```suggestion
  frontier (thickening δ E) ⊆ {x : α | inf_edist x E = ennreal.of_real δ} :=
```
(Our convention for sets is no space after or before the braces). Same thing several times below (and above?)